tempest/api/workloadmgr/barbican/test_barbican.py

```diff
@@ -2833,3 +2833,142 @@ def test_8_barbican(self):
+                volume_type_id=vol_type_id)
+
+            #Attach volume to vm...
+            self.attach_volume(volume_id, vm_id)
+
+            #create a workload with encryption status as disabled and try to attach it to vm...
+            try:
+                wid = self.workload_create([vm_id],
+                                                tvaultconf.workload_type_id, encryption=False,
+                                                workload_cleanup=True)
+
+            except Exception as e:
+                LOG.error("Exception: " + str(e))
+                reporting.add_test_step("Unencrypted Workload creation with unencrypted volume is failed to create.", tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
+
+            #create snapshot of workload created...
+            snapshot_id = self.workload_snapshot(wid, True)
+
+            self.wait_for_workload_tobe_available(wid)
+            snapshot_status = self.getSnapshotStatus(wid, snapshot_id)
+
+            if(snapshot_status == "available"):
+                LOG.debug("Full snapshot created...")
+            else:
+                reporting.add_test_step("Full snapshot is not created...", tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
+                raise Exception("Create full snapshot failed")
+
+            #now create another vm with encrypted volume...
+            #create another key pair...
+            kp1 = self.create_key_pair(tvaultconf.key_pair_name)
+
+            #create another vm...
+            vm_id1 = self.create_vm(key_pair=kp1)
+
+            #find volume_type = luks. So that existing encrypted volume type can be used.
+            #Get the volume_type_id
+            vol_type_id = -1
+            for vol in CONF.volume.volume_types:
+                if (vol.lower().find("luks") != -1):
+                    vol_type_id = CONF.volume.volume_types[vol]
+
+            if (vol_type_id == -1):
+                reporting.add_test_step("No encrypted volume available to attach", tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
+                raise Exception("No luks volume found to create encrypted volume.")
+
+            #Now create volume with derived volume type id...
+            volume_id1 = self.create_volume(
+                volume_type_id=vol_type_id)
+
+            #Attach volume to vm...
+            self.attach_volume(volume_id1, vm_id1)
+
+            #now edit the workload with workload_modify cli and try to attach vm with encrypted volume...
+            try:
+                workload_modify_command = command_argument_string.workload_modify + "--instance instance-id=" + \
+                    str(vm_id1) + " --instance instance-id=" + str(vm_id) + " " + str(wid)
+
+                rc = cli_parser.cli_returncode(workload_modify_command)
+                if rc != 0:
+                    LOG.error("Command did not execute correctly")
```

Comment:
Please capture the error thrown by command itself here. It would be helpful for verification as well.